### PR TITLE
Allow list of multiple architectures

### LINKED ISSUE #10674: Support for tasks running on more than 1 architecture
**Impact of the new feature**
This feature is particularly important for running on large HPC systems with architecture != from x86_64, since they will typically be places where no Rucio maintained storage ill be present. Hence, we have the need to be able and run a part of the task (the CPU intensive) on the HPC, and IO intensive parts (merge, cleanup, logcollect) on the site with storage, which will be typically a standard x86_64.

The idea is to submit a task specifying a LIST of architectures it can run on, and then use glidein local matching to accept / refuse parts of the task (for example, the HPC site will veto Merge, Cleanup, logcollect, while the x86 with storage will accept only these).

What is needed and already tested to some extent (see https://indico.cern.ch/event/1049288/) is the ability to
- prepare a classad matching more than one arch
- reconfiguring the payload at runtime from the local environment
- (filter the tipe of subtask -- which is not a WMA feature but a condor one)

**Is your feature request related to a problem? Please describe.**
Not now, but it will be as soon as CMS will have available multiple architecture via (for example) HPC grants

**Describe the solution you'd like**
see https://indico.cern.ch/event/1049288 for a sketched solution.
in principle, the only part missing is the interface by which the user defines the list of possible architectures for a task

**Describe alternatives you've considered**
none (but happy to hear from you!)

### PR DESCRIPTION
**Pull request by khurtado**
Fixes #10674

#### Status
In development

#### Description
- Modify condor requirements to allow multiple architectures
- Modify Scram class so that  executors match the target architecture
#### Is it backward compatible (if not, which system it affects?)
YES

### CODE REVIEW

**Comment by khurtado** [comment]
CMSSW and Python look healthy:

```
Local time: Sun Jul 31 16:29:24 CEST 2022
Hostname:   r234n01.m100.cineca.it
System:     Linux r234n01 4.18.0-147.13.2.el8_1.ppc64le #1 SMP Wed May 13 15:23:36 UTC 2020 ppc64le ppc64le ppc64le GNU/Linux
Arguments:  spiga_TC_SLC7_Marconi_TB_CMS_Marconi_220731_160802_6847-Sandbox.tar.bz2 977 0

2022-07-31 16:29:26,695:INFO:Startup:Python path: ['/m100_scratch/userexternal/dspiga00/glidein_work_area_nogpu/glide_bcIp0L/execute/dir_87019/job', '/cvmfs/cms.cern.ch/slc7_ppc64le_gcc820/external/py2-future/0.18.2/lib/python3.8/site-packages', '/m100_scratch/userexternal/dspiga00/glidein_work_area_nogpu/glide_bcIp0L/execute/dir_87019/job/WMCore.zip', '/m100_scratch/userexternal/dspiga00/glidein_work_area_nogpu/glide_bcIp0L/execute/dir_87019/job', '/cvmfs/cms.cern.ch/slc7_ppc64le_gcc820/external/python3/3.8.2/lib/python38.zip', '/cvmfs/cms.cern.ch/slc7_ppc64le_gcc820/external/python3/3.8.2/lib/python3.8', '/cvmfs/cms.cern.ch/slc7_ppc64le_gcc820/external/python3/3.8.2/lib/python3.8/lib-dynload', '/cvmfs/cms.cern.ch/slc7_ppc64le_gcc820/external/python3/3.8.2/lib/python3.8/site-packages']


2022-07-31 16:29:31,069:INFO:CMSSW:    Invoking command:
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH
/cvmfs/cms.cern.ch/slc7_ppc64le_gcc820/external/python3/3.8.2/bin/python3 -m WMCore.WMRuntime.ScriptInvoke WMTaskSpace.cmsRun1 SetupCMSSWPset 
```

I have squashed the commits and I think this is ready for reviewe

File: test/python/WMCore_t/BossAir_t/BasePlugin_t.py [hunk 1]
```diff
@@ -66,10 +66,15 @@ def testScramArchtoRequiredArch(self):
             bp.scramArchtoRequiredArch("slc7_BLAH_gcc700")
 
         self.assertEqual(bp.scramArchtoRequiredArch(['slc5_amd64_gcc481', 'slc6_amd64_gcc630']), 'X86_64')
-        self.assertEqual(bp.scramArchtoRequiredArch(['slc7_amd64_gcc10', 'slc7_aarch64_gcc700']), 'X86_64')
-        self.assertEqual(bp.scramArchtoRequiredArch(
-                ['slc7_amd64_gcc10', 'slc7_aarch64_gcc700', 'slc7_ppc64le_gcc9']), 'X86_64')
-        self.assertEqual(bp.scramArchtoRequiredArch(['slc7_aarch64_gcc700', 'slc7_ppc64le_gcc9']), 'ppc64le')
+        test = (bp.scramArchtoRequiredArch(['slc7_amd64_gcc10', 'slc7_aarch64_gcc700'])).split(',')
```
Comment (amaltaro) [inline test/python/WMCore_t/BossAir_t/BasePlugin_t.py:69]: I think one pair of parenthesis here is not really needed (the first one).

**Review by amaltaro — CHANGES_REQUESTED** [review]
Thanks Kenyi, it's looking good for me. I have left a few comments along the code for your consideration as well. In addition to those, can you please check the pycodestyle report for test/python/WMCore_t/BossAir_t/BasePlugin_t.py (there are a few new items reported here).

File: test/python/WMCore_t/BossAir_t/BasePlugin_t.py (hunk 1 above)
Comment (khurtado) [reply test/python/WMCore_t/BossAir_t/BasePlugin_t.py:69]: I think I got an error with ` .spllit()` without the parenthesis

**Review by amaltaro — APPROVED** [review]
It looks good to me, thanks Kenyi.

@belforte FYI, in case you are using one of these APIs mapping ScramArch to OS/Arch.

**Pull request merged**